app/controllers/api/v2/providers_controller.rb

```diff
@@ -41,7 +41,15 @@ def publish
         authorize @provider, :publish?
 
         if @provider.publishable?
-          head :ok
+          @provider.publish_enrichment(@current_user)
+
+          if has_synced?
+            head :ok
+          else
+            raise RuntimeError.new(
```

Comment:
can we have this instead
``` ruby
         raise RuntimeError.new(
            "#{@provider} failed to sync these courses #{@provider.syncable_courses.pluck(:course_code)}"
          )```

Reply:
Discussed in person with @defong, probably not worth this change ... we have other things to improve on:

- `Provider#to_s` should include the recruitment year
- `SearchAndCompareAPIService` should allow us to inspect the result to see what status code we got back, not only rely on a boolean

Reply:
https://trello.com/c/usEjgTt1/1877-add-recruitment-year-and-provider-code-to-coursetos

https://trello.com/c/kVWN5rBS/1876-add-recruitment-year-to-providertos